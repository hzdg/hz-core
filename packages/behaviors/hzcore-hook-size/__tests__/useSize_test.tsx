/* eslint-env jest, browser */
import React from 'react';
import {render, getByTestId} from 'react-testing-library';
import useSize, {Size} from '../src';

const SizeUser = (): JSX.Element => {
  const [size, ref] = useSize();
  return (
    <div ref={ref}>
      <pre data-testid="sizeState">{JSON.stringify(size, null, 2)}</pre>
    </div>
  );
};

const getSize = (container: HTMLElement): Size =>
  JSON.parse(getByTestId(container, 'sizeState').innerHTML);

test('useSize gets the initial size', async () => {
  const {container} = render(<SizeUser />);
  expect(container).toBeInTheDocument();
  expect(getSize(container)).toMatchObject({
    x: null,
    y: null,
    top: null,
    right: null,
    bottom: null,
    left: null,
    width: null,
    height: null,
    vw: null,
    vh: null,
  });
});
