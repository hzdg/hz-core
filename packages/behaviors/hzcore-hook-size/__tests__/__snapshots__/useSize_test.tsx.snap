// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useSize gets the initial size 1`] = `
<div
  data-testid="size"
  style="width: 120px; height: 120px;"
>
  {
  "x": 0,
  "y": 0,
  "top": 0,
  "right": 120,
  "bottom": 120,
  "left": 0,
  "width": 120,
  "height": 120,
  "vw": 0.12,
  "vh": 0.16,
  "borderBoxSize": {
    "inlineSize": 120,
    "blockSize": 120
  },
  "contentBoxSize": {
    "inlineSize": 120,
    "blockSize": 120
  }
}
</div>
`;

exports[`useSize passes the initial size to a handler 1`] = `
Array [
  Array [
    Object {
      "borderBoxSize": Object {
        "blockSize": 120,
        "inlineSize": 120,
      },
      "bottom": 120,
      "contentBoxSize": Object {
        "blockSize": 120,
        "inlineSize": 120,
      },
      "height": 120,
      "left": 0,
      "right": 120,
      "top": 0,
      "vh": 0.16,
      "vw": 0.12,
      "width": 120,
      "x": 0,
      "y": 0,
    },
  ],
]
`;

exports[`useSize uses an existing ref 1`] = `
<div
  data-testid="size"
  style="width: 120px; height: 120px;"
>
  {
  "x": 0,
  "y": 0,
  "top": 0,
  "right": 120,
  "bottom": 120,
  "left": 0,
  "width": 120,
  "height": 120,
  "vw": 0.12,
  "vh": 0.16,
  "borderBoxSize": {
    "inlineSize": 120,
    "blockSize": 120
  },
  "contentBoxSize": {
    "inlineSize": 120,
    "blockSize": 120
  }
}
</div>
`;

exports[`useSize uses an existing ref and a handler 1`] = `
Array [
  Array [
    Object {
      "borderBoxSize": Object {
        "blockSize": 120,
        "inlineSize": 120,
      },
      "bottom": 120,
      "contentBoxSize": Object {
        "blockSize": 120,
        "inlineSize": 120,
      },
      "height": 120,
      "left": 0,
      "right": 120,
      "top": 0,
      "vh": 0.16,
      "vw": 0.12,
      "width": 120,
      "x": 0,
      "y": 0,
    },
  ],
]
`;
